packages/rum/src/domain/record/record.ts

```diff
@@ -15,22 +16,29 @@ import type {
 import { RecordType, IncrementalSource } from '../../types'
 import { serializeDocument, SerializationContextStatus } from './serialize'
 import { initObservers } from './observers'
+import type { InputCallback } from './observers'
 
-import { MutationController } from './mutationObserver'
 import { getVisualViewport, getScrollX, getScrollY } from './viewports'
 import { assembleIncrementalSnapshot } from './utils'
 import { createElementsScrollPositions } from './elementsScrollPositions'
+import { withShadowDomHelpers } from './shadowDom'
 
 export interface RecordOptions {
   emit?: (record: BrowserRecord) => void
   configuration: RumConfiguration
   lifeCycle: LifeCycle
 }
 
+interface ShadowDomCallBacks {
+  stop: () => void
+  flush: () => void
+}
 export interface RecordAPI {
   stop: () => void
   takeSubsequentFullSnapshot: (timestamp?: TimeStamp) => void
+  // the following is only used for testing purposes
   flushMutations: () => void
+  shadowDomCallBacks: Map<ShadowRoot, ShadowDomCallBacks>
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** flushMutations is still used in prod code‏
```suggestion
  flushMutations: () => void
   // the following is only used for testing purposes
  shadowDomCallBacks: Map<ShadowRoot, ShadowDomCallBacks>
```